apps/sms/js/activity_handler.js

```diff
@@ -229,25 +229,24 @@ var ActivityHandler = {
     //Validate if message still exists before opening message thread
     //See issue https://bugzilla.mozilla.org/show_bug.cgi?id=837029
     if (!message) {
-      return;
+      return Promise.reject(new Error('Message is not defined!'));
```

Comment:
I think I'd simply return a Promise.resolve(). Returning a rejected promise is just like throwing an error. And I don't think it's appropriate here. However we might want to have a console.error.